File: src/pentagon.kt
import kotlin.random.Random
import kotlin.math.min
import kotlinx.coroutines.delay
import kotlinx.coroutines.runBlocking



fun main() = runBlocking {
    var progress = 0
    
    while (progress < 100) {
        val maxLimit = min(progress + 20, 100)
        progress = Random.nextInt(progress + 1, maxLimit + 1)
        
        println("Pentagon hack progress: $progress%")
        
        delay(1000)
    }
    
    if (Random.nextInt(1, 31) > 20) {
        println("Pentagon hack: Completed successfully.")
        
    } else {
        println("Pentagon hack: Failed.")
    }
}

File: src/pentagon.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	progress := 0

	for progress < 100 {
		maxLimit := progress + 20
		if maxLimit > 100 {
			maxLimit = 100
		}
		progress = rand.Intn(maxLimit-progress) + progress + 1
		fmt.Printf("Pentagon hack progress: %d%%\n", progress)

		time.Sleep(1 * time.Second)
	}

	if rand.Intn(30)+1 > 20 {
		fmt.Println("Pentagon hack: Completed successfully.")
	} else {
		fmt.Println("Pentagon hack: Failed.")
	}
}
